src/arithmetic/algebraic_expression/algebraic_expression_optimization.c

```diff
@@ -236,8 +237,36 @@ static void _AlgebraicExpression_FlattenMultiplications(AlgebraicExpression *roo
 }
 
 //------------------------------------------------------------------------------
-// Transpose pushdown
+// Transpose pushdown and in-place matrix replacement
 //------------------------------------------------------------------------------
+// Transpose an operand matrix and update the expression accordingly.
+static void _AlgebraicExpression_TransposeOperand(AlgebraicExpression *operand) {
+	if(operand->operand.diagonal == true) return;  // No need to transpose diagonal matrix.
+	GrB_Type type;
+	GrB_Index nrows;
+	GrB_Matrix replacement;
+	// Create a new empty matrix with the type and dimensions of the original.
+	GrB_Matrix_nrows(&nrows, operand->operand.matrix);
```

Comment:
Just to be more generic please call `GrB_Matrix_ncols` and use both `nrows` and `ncols`